File: pythia-0.8/packages/journal/tests/libjournal/debug.cc
// -*- C++ -*-
//
// ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
//
//                             Michael A.G. Aivazis
//                      California Institute of Technology
//                      (C) 1998-2005  All Rights Reserved
//
// <LicenseText>
//
// ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
//

#include <portinfo>
#include "journal/debug.h"
#include <iostream>

int main() {

    journal::debug_t debug("test");
    debug.activate();
    std::cout << debug.name() << ": state=" << debug.state() << std::endl;

    debug << journal::at(__HERE__) << "Hello world!" << journal::endl;

    return 0;
}

// version
// $Id: debug.cc,v 1.1.1.1 2005/03/08 16:13:54 aivazis Exp $

// End of file 

File: pythia-0.8/examples/opaldemo/cgi/debug.py
#!/usr/bin/env python
#
# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
#
#                             Michael A.G. Aivazis
#                      California Institute of Technology
#                      (C) 1998-2005  All Rights Reserved
#
# {LicenseText}
#
# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
#


def main():


    from opaldemo.applications.WebApplication import WebApplication


    class DemoApp(WebApplication):


        def main(self, *args, **kwds):
            self.render(self.retrievePage("debug"))
            return


        def __init__(self):
            WebApplication.__init__(self, name='main', asCGI=True)
            return


    app = DemoApp()
    return app.run()


# main
if __name__ == '__main__':
    # invoke the application shell
    main()


# version
__id__ = "$Id: debug.py,v 1.1 2005/03/27 15:13:06 aivazis Exp $"

# End of file 
